sound/soc/sof/intel/hda-ipc.c

```diff
@@ -116,6 +116,12 @@ void hda_dsp_ipc_get_reply(struct snd_sof_dev *sdev)
 	spin_unlock_irqrestore(&sdev->ipc_lock, flags);
 }
 
+static bool hda_dsp_ipc_is_sof(uint32_t msg)
+{
+	return (msg & (HDA_DSP_IPC_PURGE_FW | 0xf << 9)) != msg ||
```

Comment:
@lyakh maybe we can be precise in this check. If you look in hda-loader.c the cl_stream_prepare() method returns the stream tag the code loader will be assigned. So we can just check for that one instead of checking for all of them?

Reply:
@ranj063 how would you know in this context which of the tags to look for?

Reply:
@plbossart i was under the impression that the  ROM message is only received after the code loader DMA is complete and the DSP sends back the reply. The stream tag for the code loader is assigned just prior to this. Maybe I am mistaken?

Reply:
On the one hand I agree that it's good to be precise, if we *really* can be sure which tag should be coming we can check just for that one. On the other hand it looks like so far we weren't very successful with being precise here. Do you know of any drawbacks from checking for all tags here?